frontend/src/components/managePermissions/ManagePermissions.vue

```diff
@@ -56,6 +58,13 @@ const props = defineProps({
     },
 });
 
+// get user access ids from router query parameters
+const { query } = useRoute();
```

Comment:
and ManagePermissions is the parent component of the dashboard router, so the router query parameter should be received at this level, and then pass to child components, instead of reading from child components directly, even though read from child components will also work, but that's not the correct logic.

Reply:
Using useRoute() is good too; just a comment, not to request for a change. 
Using Props, even on the this parent component feels the component would be more stand-alone itself and will be easy to plug and play and probably also will be easier for testing.

Reply:
yeah I agree, updated to use Props